src/zip_api.cc

```diff
@@ -9,51 +9,59 @@ namespace {
 
 #define THROW_BAD_ARGS(msg)    \
     do {                       \
-       NanThrowTypeError(msg); \
-       NanReturnUndefined();   \
+       Nan::ThrowTypeError(msg); \
+       info.GetReturnValue().Set(Nan::Undefined()); \
```

Comment:
The macro `THROW_BAD_ARGS` should be end by a returned statement, otherwise, the code logic will continue running. The `info.GetReturnValue().Set(Nan::Undefined());` is not a return statement here.

I think we can remove the macro here, and use the following code 
```
Nan::ThrowTypeError("Bad arguments");
return
```